yellow_team.py

```diff
@@ -1,71 +1,199 @@
+from __future__ import annotations
 import json
 import random
 import socket
-
 from scipy.spatial import distance
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
 
 client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  # socket initialization
 client.connect(('127.0.0.1', 7976))  # connecting client to server
 
+class Handler(ABC):
+    @abstractmethod
+    def set_next(self, handler: Handler):
+        pass
+
+    @abstractmethod
+    def handle(self, request):
+        pass
+
+
+class AbstractHandler(Handler):
+    _next_handler: Handler = None
+
+
+    def set_next(self, handler: Handler):
```

Comment:
jól látom h ez kopi paszta? A mintákat pont azért hozták létre h javítsák a kód skálázhatóságát és te duplázol kódot :D

Reply:
teljesen jól látod :D még csak most ismerkedem ezekkel kell a mankó :D

Reply:
itt kitöröltema HANDLER osztályt és a ABSTRACHANDLER osztály lett ABS, gondolom erre gondoltál

Reply:
blue és yellow file-okban van a másolat.

Reply:
akkor a Handler(ABS) kell? hát igen, mivel két kliens a blue meg a yellow, így ha külön gépen lennének kell mindkettőnek a kód

Reply:
de csak azért mert 2 gépen kell a kód a classt még nem kell másolni.

Reply:
akkor ilyenkor hogy kéne csinálni?

Reply:
rakd ki az implementációt egy fájlba és azt is másold a játékkal.